src/shared/bootspec.c

```diff
@@ -243,6 +245,115 @@ static int parse_tries(const char *fname, const char **p, unsigned *ret) {
         return 1;
 }
 
+static int parse_profile_from_options(const char *options, unsigned *ret_profile, const char **ret_tail) {
+        unsigned u = 0;
+
+        assert(options);
+
+        if (options[0] != '@')
+                return 0;
+
+        size_t nr_of_digits = strspn(options + 1, DIGITS);
+        if (nr_of_digits == 0)
+                return 0;
+
+        size_t prefixlen = nr_of_digits + 1;
+        if (options[prefixlen] == '\0') {
+                if (safe_atou_full(options + 1, 10, &u) < 0)
+                        return 0;
+        } else if (options[prefixlen] == ' ') {
+                prefixlen++;
+
+                _cleanup_free_ char *profile = strndup(options + 1, nr_of_digits);
+                if (!profile)
+                        return log_oom();
+
+                if (safe_atou_full(profile, 10, &u) < 0)
+                        return 0;
+        } else
+                return 0;
+
+        if (ret_profile)
+                *ret_profile = u;
+        if (ret_tail)
+                *ret_tail = options + prefixlen;
+
+        return 1;
+}
+
+static int set_profile_from_options(char ***options, unsigned *profile) {
+        int r;
+
+        assert(options);
+        assert(profile);
+
+        if (!*options)
+                return 0;
+
+        assert(**options);
+
+        unsigned u = *profile;
+        const char *tail = NULL;
+        r = parse_profile_from_options(**options, &u, &tail);
+        if (r <= 0)
+                return r;
+
+        /* If options consisted only of the profile selector ('@X'), free it */
+        if (strv_length(*options) == 1 && *tail == 0)
+                *options = strv_free(*options);
+
```

Comment:
please remove this empty line (i.e. we try to keep if/else branches together unless they are contain multiline options